src/util/promise.ts

```diff
@@ -27,13 +29,16 @@ export async function runSequentially<T>(promisers: Promiser<T>[]): Promise<T[]>
     } catch (err) {
       const errors = err as any;
 
+      if (errors.error) {
+        result.push(errors.error);
+      }
+
       if (errors.originalError) {
         result.push(errors.originalError);
-      } else if (errors.error) {
-        result.push(errors.error);
-      } else {
-        throw err;
       }
+
+      result.push(errors);
+      throw flatten(result.filter(r => !isNil(r)));
```

Comment:
Maybe separate out the results and errors array here. Making the result and errors array the same is confusing and not readable.